wolfssl/wolfcrypt/types.h

```diff
@@ -712,6 +712,105 @@
         #define WC_NORETURN
     #endif
 
+    #ifdef HAVE_PK_CALLBACKS
+        struct WOLFSSL;
+        #ifdef HAVE_ECC
+            typedef int (*CallbackEccSign)(struct WOLFSSL* ssl,
```

Comment:
Having the WOLFSSL pointer at the wolfCrypt level should be avoided.  Is there another way?

Reply:
I am kind of stumped here. This was moved from ssl.h to avoid having wolfCrypt components in wolfSSL. This is an opaque structure reference, similar to the use in ```wolfevent.h```. While we could change the callback declaration to pass a void pointer instead, eventually we will have to cast it as a WOLFSSL structure. I am very open to suggestions.